src/http/controllers/millions.controller.ts

```diff
@@ -246,6 +251,63 @@ export class MillionsController {
         });
     }
 
+    @ApiOkResponse({ status: 200, type: MillionsCampaignResponse })
+    @Get('campaigns')
+    async campaigns(@Req() request: ExplorerRequest): Promise<DataResponse> {
+        const [campaigns, total] = await this._millionsCampaignService.fetch(request.pagination.skip, request.pagination.limit);
+
+        return new DataResponse({
+            result: campaigns.map((campaign) => plainToInstance(MillionsCampaignResponse, campaign)),
+            metadata: new DataResponseMetadata({
+                page: request.pagination.page,
+                limit: request.pagination.limit,
+                items_count: campaigns.length,
+                items_total: total,
+            }),
+        });
+    }
+
+    @ApiOkResponse({ status: 200, type: MillionsCampaignResponse })
+    @Get('campaigns/:id')
+    async campaignById(@Param('id') id: string): Promise<DataResponse> {
+        const campaign = await this._millionsCampaignService.getById(id);
+
+        if (!campaign) {
+            throw new NotFoundException('Campaign not found');
+        }
+
+        return new DataResponse({
+            result: plainToInstance(MillionsCampaignResponse, campaign),
+        });
+    }
+
+    @ApiOkResponse({ status: 200, type: MillionsCampaignResponse })
+    @Post('campaigns/participate')
+    async participateCampaign(@Req() request: MillionsCampaignParticipationRequest): Promise<DataResponse> {
+        const campaign = await this._millionsCampaignService.getById(request.campaign_id);
+
+        if (!campaign) {
+            throw new NotFoundException('Campaign not found');
+        }
+
+        if (!(await bcrypt.compare(request.password, campaign.password))) {
+            throw new NotFoundException('Password is incorrect');
+        }
+
+        if (await this._millionsCampaignMemberService.getByCampaignIdAndWalletAddress(request.campaign_id, request.wallet_address)) {
+            throw new NotFoundException('You already participated in this campaign');
```

Comment:
NotFoundException is semantically confusing here as technically the ressource exists, but shouldn't be used.
A ConflicException could be used here, or alternatively a standard new Error()